flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/strategy/ConsumedPartitionGroup.java

```diff
@@ -132,4 +133,25 @@ public boolean areAllPartitionsFinished() {
     public ResultPartitionType getResultPartitionType() {
         return resultPartitionType;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        ConsumedPartitionGroup that = (ConsumedPartitionGroup) o;
+        return numConsumers == that.numConsumers
+                && Objects.equals(resultPartitions, that.resultPartitions)
+                && Objects.equals(intermediateDataSetID, that.intermediateDataSetID)
+                && resultPartitionType == that.resultPartitionType;
+    }
```

Comment:
What if there are two `ConsumedPartitionGroup` that are "equal" while having different `unfinishedPartitions`? Which one should be used?

We probably want to make sure for each group there's at most one `ConsumedPartitionGroup` object. Thus, `this` and `o` are equal only if they are the same object.

Reply:
Make sense, fixed it.